Packs/SOCRadar/Integrations/SOCRadarIncidents/SOCRadarIncidents.py

```diff
@@ -0,0 +1,542 @@
+        message = f"SOCRadar API Response: {raw_response.get('message', '')}"
+    else:
+        message = f"Error while getting API response. SOCRadar API Response: {raw_response.get('message', '')}"
+        raise DemistoException(message=message)
+
+    return CommandResults(
+        readable_output=message,
+        raw_response=raw_response
+    )
+
+
+''' MAIN FUNCTION '''
+
+
+def main() -> None:
+    """main function, parses params and runs command functions
+
+    :return:
+    :rtype:
+    """
+
+    api_key = demisto.params().get('apikey')
+    socradar_company_id = demisto.params().get('socradar_company_id')
+
+    base_url = SOCRADAR_API_ENDPOINT
+    first_fetch_time = arg_to_datetime(
+        arg=demisto.params().get('first_fetch', '40 days').strip(),
+        arg_name='First fetch time',
+        required=True
+    )
+
+    first_fetch_timestamp = int(first_fetch_time.timestamp()) if first_fetch_time else 0
+    assert isinstance(first_fetch_timestamp, int)
+
+    verify_certificate = argToBoolean(demisto.params().get('insecure', False))
+    proxy = demisto.params().get('proxy', False)
+
+    demisto.debug(f'Command being called is {demisto.command()}')
+    try:
+
+        client = Client(
+            base_url=base_url,
+            api_key=api_key,
+            socradar_company_id=socradar_company_id,
+            verify=verify_certificate,
+            proxy=proxy)
+        command = demisto.command()
+
+        if command == 'test-module':
+            return_results(test_module(client))
+        elif command == 'fetch-incidents':
+            resolution_status = demisto.params().get('resolution_status')
+            fp_status = demisto.params().get('fp_status')
+            incident_main_type = demisto.params().get('incident_main_type')
+            incident_sub_type = demisto.params().get('incident_sub_type')
+            severity = demisto.params().get('severity')
+
+            max_results = arg_to_number(
+                arg=demisto.params().get('max_fetch'),
+                arg_name='max_fetch',
+                required=False
+            )
+            if not max_results or max_results > MAX_INCIDENTS_TO_FETCH:
+                max_results = MAX_INCIDENTS_TO_FETCH
+
+            next_run, incidents = fetch_incidents(
+                client=client,
+                max_results=max_results,
+                last_run=demisto.getLastRun(),  # getLastRun() gets the last run dict
+                first_fetch_time=first_fetch_timestamp,
+                resolution_status=resolution_status,
+                fp_status=fp_status,
+                severity=severity,
+                incident_main_type=incident_main_type,
+                incident_sub_type=incident_sub_type,
+            )
+
+            demisto.setLastRun(next_run)
+            demisto.incidents(incidents)
+
+        elif command == 'socradar-mark-incident-fp':
+            incident_id = demisto.args().get('socradar_incident_id')
```

Comment:
Please move the arguments to the command itself (like you did in the ThreatFusion integration

Reply:
done